src/classes/RD_RecurringDonations.cls

```diff
@@ -1214,4 +1279,137 @@ public class RD_RecurringDonations {
+
+        if (newRD.npe03__Organization__c != oldRD.npe03__Organization__c
+                || newRD.npe03__Contact__c != oldRD.npe03__Contact__c
+                || newRD.npe03__Amount__c != oldRD.npe03__Amount__c
+                || newRD.npe03__Installment_Period__c != oldRD.npe03__Installment_Period__c
+                || newRD.npe03__Installments__c != oldRD.npe03__Installments__c
+                || newRD.Always_Use_Last_Day_Of_Month__c != oldRD.Always_Use_Last_Day_Of_Month__c
+                || newRD.npe03__Schedule_Type__c != oldRD.npe03__Schedule_Type__c
+                || newRD.npe03__Open_Ended_Status__c != oldRD.npe03__Open_Ended_Status__c
+                || newRD.npe03__Next_Payment_Date__c != oldRD.npe03__Next_Payment_Date__c
+                || newRD.npe03__Recurring_Donation_Campaign__c != oldRD.npe03__Recurring_Donation_Campaign__c
+                ) {
+            return true;
+        }
+
+        for (String s : customFieldMappingSettings.keySet()) {
+            npe03__Custom_Field_Mapping__c cfm = customFieldMappingSettings.get(s);
+            if (newRD.get(cfm.npe03__Recurring_Donation_Field__c) != oldRD.get(cfm.npe03__Recurring_Donation_Field__c)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /*******************************************************************************************************
+    * @description Updates an Opportunity with given parameters
+    * @param opp The Opportunity to update
+    * @param r The parent Recurring Donation
+    * @param oppCloseDate The close date for the Opportunity
+    * @param amount The Amount of the Opportunity
+    * @param con The Contact involved, can be null
+    * @param rdcounter Opportunity counter for naming and Recurring_Donation_Installment_Number__c
+    * @return Opportunity
+    */
+    private static Opportunity normalizeOppForRD(Opportunity opp, npe03__Recurring_Donation__c r, Date oppCloseDate, Decimal amount, Contact con, Integer rdcounter) {
+
+        npe03__Recurring_Donations_Settings__c rds = UTIL_CustomSettingsFacade.getRecurringDonationsSettings();
+
+        opp.CloseDate = oppCloseDate;
+        opp.OwnerId = r.OwnerId;
+        opp.Amount = amount;
+        opp.Primary_Contact__c = r.npe03__Contact__c;
+        opp.Recurring_Donation_Installment_Number__c = rdcounter;
+        opp.StageName = System.label.npe03.RecurringDonationStageName;
+        opp.npe03__Recurring_Donation__c = r.Id;
+
+        if(isMultiCurrency){
+            //set the donation currency field to equal the recurring donation currency
+            opp.put(OppCurrencyField,r.get(RDCurrencyField));
+        }
+
+        if ( r.npe03__Organization__c != null ) {
+            opp.AccountId = r.npe03__Organization__c;
+        } else if ( con != null ) {
+            opp.AccountId = con.AccountId;
+            opp.npe01__Contact_Id_For_Role__c = con.id;
```

Comment:
Could you add Primary_Contact__c here as well?

Reply:
done